plugins/scaffolder-backend/src/scaffolder/stages/templater/cookiecutter.ts

```diff
@@ -78,8 +78,10 @@ export class CookieCutter implements TemplaterBase {
       });
     }
 
+    const [generated = ''] = await fs.readdir(resultDir);
```

Comment:
What's the reason for this change? Could use a clarifying comment why it's safe to assume this.

Also seems like defaulting to `''` is a bit strange, as that would mean the template has no output? Thinking we might wanna just error out in that case.

Reply:
From my PR description,

> I also changed how the path to the generated output is created. Instead of relying on user-specified component_id, I'm just taking the only generated directory in the template directory. This allows me to generate component_id using cookiecutter variables.

The previous implementation relied on `component-id` which was supposed to be passed from the UI. When I created the template, I used cookiecutter variables to define component-id from a name so it takes care sluggifying the component-id. Then I discovered that cookiecutter variables are not available when this service is running. As a result, the Node was throwing an error because it couldn't create a path with an `undefined` segment. 

I couldn't find a way to pull the computed values from cookiecutter. Instead, I took the only generated directory and used it as the path. (this could be incorrect but it seems like it should work reliably).

The test stubs out `runDockerContainer` which means that the result directory is never updated. As a result, `[generated] = await fs.readdir(resultDir)` is undefined. 

There are a bunch of ways to make this better,
1. Figure out a way to allow combining cookiecutter variables and values passed in from the UI
2. Refactor tests to stub out `runDockerContainer` to populate results directory
3. Add tests for `await commandExists('cookiecutter') === true` scenario

Now that I look at it, I should at least do 3. What do you want to do about 1 & 2?

Reply:
Alright, makes sense, please do add a comment explaining what's going on and why. Also likely best to throw an error if `generated` ends up being `undefined`, as there's no way that wouldn't lead to more errors later anyway, better catch that early 😅 

I think `generated` could be `subDir` or smth too?

Regarding the tests, use `mock-fs` for testing file system things. If it would lead to too much refactoring, prolly just `jest.spyOn(fs, 'readdir')`

Reply:
2 & 3 are done